src/main/java/org/opensearch/knn/index/codec/KNN910Codec/KNN910Codec.java

```diff
@@ -8,26 +8,23 @@
 import org.apache.lucene.codecs.CompoundFormat;
 import org.apache.lucene.codecs.DocValuesFormat;
 import org.apache.lucene.codecs.FilterCodec;
+import org.opensearch.knn.index.codec.KNNCodecVersion;
 import org.opensearch.knn.index.codec.KNNFormatFacade;
-import org.opensearch.knn.index.codec.KNNFormatFactory;
-
-import static org.opensearch.knn.index.codec.KNNCodecFactory.CodecDelegateFactory.createKNN91DefaultDelegate;
 
 /**
  * Extends the Codec to support a new file format for KNN index
  * based on the mappings.
  *
  */
 public final class KNN910Codec extends FilterCodec {
-
-    private static final String KNN910 = "KNN910Codec";
+    private static final KNNCodecVersion VERSION = KNNCodecVersion.V_9_1_0;
     private final KNNFormatFacade knnFormatFacade;
 
     /**
      * No arg constructor that uses Lucene91 as the delegate
      */
     public KNN910Codec() {
-        this(createKNN91DefaultDelegate());
+        this(VERSION.getDefaultCodecDelegate());
```

Comment:
Why should this not map to the delegate for 91?

Reply:
It's the same delegate for 9.1, just in different notation. VERSION points to V_9_1_0, defaultCodecDelegate is "new Lucene91Codec()", in updated version I moved this logic from KNNCodecFactory class to KNNCodecVersion.